homeassistant/components/homekit_controller/air_quality.py

```diff
@@ -0,0 +1,110 @@
+"""Support for HomeKit Controller air quality sensors."""
+from homekit.model.characteristics import CharacteristicsTypes
+
+from homeassistant.components.air_quality import AirQualityEntity
+
+from . import KNOWN_DEVICES, HomeKitEntity
+
+AIR_QUALITY_TEXT = {
+    0: "unknown",
+    1: "excellent",
+    2: "good",
+    3: "fair",
+    4: "inferior",
+    5: "poor",
+}
+
+
+class HomeAirQualitySensor(HomeKitEntity, AirQualityEntity):
+    """Representation of a HomeKit Controller Air Quality sensor."""
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity cares about."""
+        return [
+            CharacteristicsTypes.AIR_QUALITY,
+            CharacteristicsTypes.DENSITY_PM25,
+            CharacteristicsTypes.DENSITY_PM10,
+            CharacteristicsTypes.DENSITY_OZONE,
+            CharacteristicsTypes.DENSITY_NO2,
+            CharacteristicsTypes.DENSITY_SO2,
+            CharacteristicsTypes.DENSITY_VOC,
+        ]
+
+    @property
+    def particulate_matter_2_5(self):
+        """
+        Return the particulate matter 2.5 level.
+
+        This is required on the HA side but optional on the HomeKit side, so we
+        always return a value.
+        """
+        return self.get_hk_char_value(CharacteristicsTypes.DENSITY_PM25) or 0
+
+    @property
+    def particulate_matter_10(self):
+        """
+        Return the particulate matter 10 level.
+
+        This is required on the HA side but optional on the HomeKit side, so we
+        always return a value.
+        """
+        return self.get_hk_char_value(CharacteristicsTypes.DENSITY_PM10) or 0
+
+    @property
+    def ozone(self):
+        """Return the O3 (ozone) level."""
+        return self.get_hk_char_value(CharacteristicsTypes.DENSITY_OZONE)
+
+    @property
+    def sulphur_dioxide(self):
+        """Return the SO2 (sulphur dioxide) level."""
+        return self.get_hk_char_value(CharacteristicsTypes.DENSITY_SO2)
+
+    @property
+    def nitrogen_dioxide(self):
+        """Return the NO2 (nitrogen dioxide) level."""
+        return self.get_hk_char_value(CharacteristicsTypes.DENSITY_NO2)
+
+    @property
+    def air_quality_text(self):
+        """Return the Air Quality Index (AQI)."""
+        air_quality = self.get_hk_char_value(CharacteristicsTypes.AIR_QUALITY)
+        return AIR_QUALITY_TEXT.get(air_quality, "unknown")
+
+    @property
+    def volatile_organic_compounds(self):
+        """Return the volatile organic compounds (VOC) level."""
+        return self.get_hk_char_value(CharacteristicsTypes.DENSITY_VOC)
+
+    @property
+    def state_attributes(self):
```

Comment:
Don't overwrite `state_attributes`. Integration platforms should only implement `device_state_attributes`.

Reply:
Done.